service/src/main/java/bio/terra/workspace/service/iam/ProxiedAuthenticatedUserRequestFactory.java

```diff
@@ -1,26 +1,55 @@
 package bio.terra.workspace.service.iam;
 
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest.AuthType;
 import java.util.Optional;
 import javax.servlet.http.HttpServletRequest;
 import org.apache.commons.lang3.StringUtils;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProxiedAuthenticatedUserRequestFactory implements AuthenticatedUserRequestFactory {
+  private static final String BEARER = "Bearer ";
 
-  // Method to build an AuthenticatedUserRequest from data available to the controller
+  // Method to build an AuthenticatedUserRequest.
+  // We try three possible authentication methods. If none succeed, we return
+  // an empty AuthenticatedUserRequest object.
   public AuthenticatedUserRequest from(HttpServletRequest servletRequest) {
-    String token =
-        Optional.ofNullable(servletRequest.getHeader(AuthHeaderKeys.OIDC_ACCESS_TOKEN.getKeyName()))
-            .orElseGet(
-                () -> {
-                  String authHeader =
-                      servletRequest.getHeader(AuthHeaderKeys.AUTHORIZATION.getKeyName());
-                  return StringUtils.substring(authHeader, "Bearer:".length());
-                });
-    return new AuthenticatedUserRequest()
-        .email(servletRequest.getHeader(AuthHeaderKeys.OIDC_CLAIM_EMAIL.getKeyName()))
-        .subjectId(servletRequest.getHeader(AuthHeaderKeys.OIDC_CLAIM_USER_ID.getKeyName()))
-        .token(Optional.ofNullable(token));
+    return fromOidc(servletRequest)
+        .orElse(
+            fromBearer(servletRequest)
+                .orElse(
+                    new AuthenticatedUserRequest()
+                        .token(Optional.empty())
+                        .authType(AuthType.NONE)));
+  }
+
+  private Optional<AuthenticatedUserRequest> fromOidc(HttpServletRequest servletRequest) {
+    Optional<String> token =
+        Optional.ofNullable(
+            servletRequest.getHeader(AuthHeaderKeys.OIDC_ACCESS_TOKEN.getKeyName()));
+    if (token.isEmpty()) {
+      return Optional.empty();
+    }
+
+    return Optional.of(
+        new AuthenticatedUserRequest()
+            .email(servletRequest.getHeader(AuthHeaderKeys.OIDC_CLAIM_EMAIL.getKeyName()))
+            .subjectId(servletRequest.getHeader(AuthHeaderKeys.OIDC_CLAIM_USER_ID.getKeyName()))
+            .token(token)
+            .authType(AuthType.OIDC));
+  }
+
+  private Optional<AuthenticatedUserRequest> fromBearer(HttpServletRequest servletRequest) {
+    String authHeader = servletRequest.getHeader(AuthHeaderKeys.AUTHORIZATION.getKeyName());
+    if (StringUtils.startsWith(authHeader, BEARER)) {
+      return Optional.of(
+          new AuthenticatedUserRequest()
+              .email(servletRequest.getHeader(AuthHeaderKeys.OIDC_CLAIM_EMAIL.getKeyName()))
```

Comment:
Even for the `BEARER` case we expect OIDC headers to get email, subjectId claims. It seems WSM always needs to be behind OIDC auth… so is it worth having an `AuthType` enum?